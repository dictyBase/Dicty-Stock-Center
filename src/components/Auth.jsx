import React from 'react'
import { Flex } from 'rebass'
import FontAwesome from 'react-fontawesome'
import { CenteredBox } from 'styles'

const AuthLoader = () => {
  AuthLoader.displayName = 'loading component during authentication'
  return (
    <Flex wrap justify="center">
      <CenteredBox>
        <h1>Logging in...</h1>
        <br />
        <FontAwesome name="spinner" size="5x" pulse fixedWidth />
      </CenteredBox>
    </Flex>
  )
}

export default AuthLoader
