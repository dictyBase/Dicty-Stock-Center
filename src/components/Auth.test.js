import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { AuthLoader, Logout } from './Auth'

test('matching a snapshot of AuthLoader', () => {
    const component = renderer.create(<AuthLoader />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})

test('matching a snapshot of Logout', () => {
    const component = renderer.create(
    <BrowserRouter>
        <Logout />
    </BrowserRouter>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})