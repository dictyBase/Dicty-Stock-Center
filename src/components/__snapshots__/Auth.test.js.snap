// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of AuthLoader 1`] = `
<div
  className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
  is={null}
>
  <div
    className="sc-hzNEM fyqJHU sc-ifAKCX cFlEyZ sc-bdVaJa iHZvIS"
    is={null}
  >
    <h1>
      Logging in...
    </h1>
    <br />
    <span
      aria-hidden={true}
      className="fa fa-spinner fa-5x fa-pulse fa-fw"
    />
  </div>
</div>
`;

exports[`matching a snapshot of Logout 1`] = `
<div
  className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
  is={null}
>
  <div
    className="sc-hzNEM fyqJHU sc-ifAKCX cFlEyZ sc-bdVaJa csVOHI"
    is={null}
  >
    <div
      className="sc-dymIpo dmPwKo"
    >
      <span
        aria-hidden={true}
        className="fa fa-check-circle-o fa-5x"
      />
      <h1>
        Logout successful!
      </h1>
      <p>
        You have successfully logged out of Dicty Stock Center
      </p>
      <button
        className="large sc-bmyXtO dHjBJs"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Stock Center Home
        </a>
      </button>
    </div>
  </div>
</div>
`;
